src/model/user-course.model.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  JoinColumn,
+  ManyToOne,
+  Entity,
+  Column,
+  PrimaryGeneratedColumn,
+  CreateDateColumn,
+  UpdateDateColumn,
+  DeleteDateColumn,
+} from 'typeorm'
+
+import { UserCourseLevel } from 'devu-shared-modules'
+import UserModel from './users.model'
+import CourseModel from './courses.model'
+
+@Entity('user_courses')
+export default class UserCourseModel {
+  @PrimaryGeneratedColumn()
+  id: number
+
+  @CreateDateColumn({ name: 'created_at' })
+  createdAt: Date
+
+  @UpdateDateColumn({ name: 'updated_at' })
+  updatedAt: Date
+
+  @DeleteDateColumn({ name: 'deleted_at' })
+  deletedAt?: Date
+
+  // Foreign key
+  @JoinColumn({ name: 'user_id' })
+  @ManyToOne(() => UserModel)
+  userId: number
```

Comment:
Do you not need the `@Column` decorator here? Does it know to name the column correctly without it?

Reply:
It appears that it's not needed, but when it's added a NOT NULL constraint will be added to the migration. I can add it back since we'll likely want the not null